rclcpp/src/rclcpp/duration.cpp

```diff
@@ -185,9 +186,11 @@ bounds_check_duration_scale(int64_t dns, double scale, uint64_t max)
 {
   auto abs_dns = static_cast<uint64_t>(std::abs(dns));
   auto abs_scale = std::abs(scale);
-
-  if (abs_scale > 1.0 && abs_dns > static_cast<uint64_t>(max / abs_scale)) {
-    if ((dns > 0 && scale > 0) || (dns < 0 && scale < 0)) {
+  long double max_ld = static_cast<long double>(max);
```

Comment:
Do we actually need the extra precision of a `long double` here?  I'm particularly worried that this will have a performance impact because this will probably end up calling software long double routines.  Further, we already know that (in the one and only caller to this), that max is going to be `std::numeric_limites<rcl_duration_value_t>::max()`, so probably INT64_MAX, which I don't think we need to store in a `long double`.  I haven't fully thought through all of the implications here, so I'm glad to hear arguments about why a `long double` would be needed.